react-forms/src/DatePicker.tsx

```diff
@@ -80,28 +81,13 @@ export default function DateTimePicker<T>(props: IProps<T>) {
     React.useEffect(() => {
         if (!recordChange.current) return;
         const date = moment(boxRecord, boxFormat);
-        const validStartDate = moment("1753-01-01", "YYYY-MM-DD");
 
-        let valid = true;
+        validateDate(date);
```

Comment:
This returns a `boolean` but it is not used. This also causes issues because even if `validateDate` returns `false` here we still go into the `if` in Line 90